biojava-structure/src/main/java/org/biojava/nbio/structure/io/cif/CifFileSupplierImpl.java

```diff
@@ -0,0 +1,316 @@
+        return wrappedAtoms;
+    }
+
+    static class WrappedAtom {
+        private final Chain chain;
+        private final int model;
+        private final String chainName;
+        private final String chainId;
+        private final Atom atom;
+        private final int atomId;
+
+        public WrappedAtom(Chain chain, int model, String chainName, String chainId, Atom atom, int atomId) {
+            this.chain = chain;
+            this.model = model;
+            this.chainName = chainName;
+            this.chainId = chainId;
+            this.atom = atom;
+            this.atomId = atomId;
+        }
+
+        public Chain getChain() {
+            return chain;
+        }
+
+        public int getModel() {
+            return model;
+        }
+
+        public String getChainName() {
+            return chainName;
+        }
+
+        public String getChainId() {
+            return chainId;
+        }
+
+        public Atom getAtom() {
+            return atom;
+        }
+
+        public int getAtomId() {
+            return atomId;
+        }
+    }
+
+    private static Collector<WrappedAtom, ?, Category> toAtomSite() {
+        return Collector.of(AtomSiteCollector::new,
+                AtomSiteCollector::accept,
+                AtomSiteCollector::combine,
+                AtomSiteCollector::get);
+    }
+
+    static class AtomSiteCollector implements Consumer<WrappedAtom> {
+        private final StrColumnBuilder<CategoryBuilder> groupPDB;
+        private final IntColumnBuilder<CategoryBuilder> id;
+        private final StrColumnBuilder<CategoryBuilder> typeSymbol;
+        private final StrColumnBuilder<CategoryBuilder> labelAtomId;
+        private final StrColumnBuilder<CategoryBuilder> labelAltId;
+        private final StrColumnBuilder<CategoryBuilder> labelCompId;
+        private final StrColumnBuilder<CategoryBuilder> labelAsymId;
+        private final StrColumnBuilder<CategoryBuilder> labelEntityId;
+        private final IntColumnBuilder<CategoryBuilder> labelSeqId;
+        private final StrColumnBuilder<CategoryBuilder> pdbxPDBInsCode;
+        private final FloatColumnBuilder<CategoryBuilder> cartnX;
+        private final FloatColumnBuilder<CategoryBuilder> cartnY;
+        private final FloatColumnBuilder<CategoryBuilder> cartnZ;
+        private final FloatColumnBuilder<CategoryBuilder> occupancy;
+        private final FloatColumnBuilder<CategoryBuilder> bIsoOrEquiv;
+        private final IntColumnBuilder<CategoryBuilder> authSeqId;
+        private final StrColumnBuilder<CategoryBuilder> authCompId;
+        private final StrColumnBuilder<CategoryBuilder> authAsymId;
+        private final StrColumnBuilder<CategoryBuilder> authAtomId;
+        private final IntColumnBuilder<CategoryBuilder> pdbxPDBModelNum;
+
+        AtomSiteCollector() {
+            // TODO this doesn't really make the case for the builder ;)
+            this.groupPDB = new StrColumnBuilder<>("atom_site", "group_PDB", null);
```

Comment:
I thought these hard-coded strings shouldn't be needed. They should come from metadata available via the CIFTools library. Or am I missing something?

Reply:
Thanks for pointing this out. 196e0c5 replaces the hard-coded category and column names with the corresponding builder functions.